plugins/infrastructure/tools/mcp-adapter.py

```
@@ -0,0 +1,352 @@
+            return [inputs["script_file"]] + raw
+
+    elif tool in ("openroad", "opensta"):
+        if "tcl_script" in inputs:
+            tf = tempfile.NamedTemporaryFile(
+                mode="w", suffix=".tcl", delete=False, prefix="mcp_tcl_"
+            )
+            tf.write(inputs["tcl_script"])
+            tf.flush()
+            tf.close()
+            print(f"[mcp-adapter] wrote Tcl script to {tf.name}", file=sys.stderr)
+            return [tf.name] + raw
```

Ruff (0.15.10)

[warning] Consider `[tf.name, *raw]` instead of concatenation

Replace with `[tf.name, *raw]`

(RUF005)